src/Sulu/Bundle/HttpCacheBundle/DependencyInjection/SuluHttpCacheExtension.php

```diff
@@ -28,117 +28,93 @@ class SuluHttpCacheExtension extends Extension implements PrependExtensionInterf
+    public function getConfiguration(array $config, ContainerBuilder $container)
     {
-        $config = $this->processConfiguration($this->getConfiguration($configs, $container), $configs);
-
-        $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
-        $loader->load('event-subscribers.xml');
-        $loader->load('proxy-client.xml');
-        $loader->load('structure-cache-handlers.xml');
-        $loader->load('services.xml');
-
-        $this->configureProxyClient($config['proxy_client'], $container);
-        $this->configureStructureCacheHandlers($config['handlers'], $container);
-
-        $container->setAlias('sulu_http_cache.handler', 'sulu_http_cache.handler.aggregate');
+        return new Configuration($container->getParameter('kernel.debug'));
     }
 
     /**
-     * Configure the proxy client services.
-     *
-     * @param array $config
-     * @param ContainerBuilder $container
+     * {@inheritdoc}
      */
-    private function configureProxyClient($config, ContainerBuilder $container)
+    public function load(array $configs, ContainerBuilder $container)
     {
-        $this->configureProxyClientVarnish($config['varnish'], $container);
-
-        // get default
-        $proxyClientName = null;
-        foreach ($config as $name => $proxyClient) {
-            if ($proxyClient['enabled'] === false) {
-                continue;
-            }
+        $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__ . '/../Resources/config'));
+        $loader->load('event-subscribers.xml');
+        $loader->load('services.xml');
 
-            if (null !== $proxyClientName) {
-                throw new InvalidConfigurationException(
-                    sprintf(
-                        'Cannot enable more than one proxy, trying to enable "%s" when "%s" is already enabled',
-                        $name,
-                        $proxyClientName
-                    )
-                );
-            }
+        $configuration = $this->getConfiguration($configs, $container);
+        $config = $this->processConfiguration($configuration, $configs);
 
-            $proxyClientName = $name;
+        if (!$this->shouldCache($container)) {
+            return;
         }
 
-        if (null === $proxyClientName) {
-            $proxyClientName = 'symfony';
+        $proxyClientAvailable = false;
+        if (array_key_exists('proxy_client', $config)) {
+            foreach ($config['proxy_client'] as $proxyClient) {
+                if (true === $proxyClient['enabled']) {
+                    $proxyClientAvailable = true;
+                    break;
+                }
+            }
         }
 
-        $container->setParameter('sulu_http_cache.proxy_client.name', $proxyClientName);
-        $container->setAlias('sulu_http_cache.proxy_client', 'sulu_http_cache.proxy_client.' . $proxyClientName);
-    }
+        if ($proxyClientAvailable) {
```

Comment:
Do you think it is necessary to load this configuration only based on the `proxyClient`? I think it doesn't hurt if these services are loaded (correct me if I'm wrong) and there is less place for potential errors if we remove that.